### ISSUE TITLE
[AU4] Clips - Pitch and speed

### ISSUE THREAD

**Issue by DilsonsPickles** [audacity/audacity#6950]
### Description

The User can adjust a clips pitch and speed via the pitch and speed dialog. Any adjustments can be rendered using the "Render pitch and speed" action that is found in the clip's context menu.

[Figma link](https://www.figma.com/design/01XLIICmy2AG6t1ieyTp66/Clips?node-id=1196-51236)

**Comment by DilsonsPickles** [comment]
@Eism please be wary that changes were made to this design due to a change in the clip visuals. 

**Here's a quick rundown of the changes:**

- Clip header colour is now always text-primary, regardless of selection state
- Clip properties hover states have changes to ensure contrast remains AAA in all interaction states

**Comment by dozzzzer** [comment]
@Eism 
####  1. When changing the clip's speed, it doesn't affect playback immediately. It has to be stopped first, and only then, after playback is re-started, you can hear the changes to the clip's speed.

#### 2. "Render pitch and speed" doesn't cause the progress dialog to show up as AU3 does. Instead, the User is facing the "busy" cursor animation and then gets the resulting rendered clip.

<img width="194" alt="image" src="https://github.com/user-attachments/assets/64bb15f7-7e06-43e0-bf5d-a350f0784ebd">

#### 3. The clip's pitch and speed value cannot be reset by left-click with the Cmd (Ctrl) key held down. 

#### 4. Changing the clip's speed or pitch isn't undoable.

#### 5. With the Pitch and Speed dialog opened up, interacting with Audacity isn't possible. Clicking outside of the dialog produces a beeping sound.

#### 6. No validation in the dialog's text boxes. You can enter 4 digits into the Speed text box and you also can enter literally anything into the Semitones/cents text boxes without them being changed to sensible values.

<img width="320" alt="image" src="https://github.com/user-attachments/assets/7c012325-0789-49c8-9a7e-3f9251337190">
Validation rules should be implemented in the same way as in AU3:
- when entering a garbage value into the text box, it must be changed to the previously entered numeric value when clicking outside of the text box;
- a maximum of 3 digits is allowed into the Speed text box.

#### 7. Changing either the clip's speed to pitch doesn't make the project dirty, i.e. when changing either of these properties of the clip and trying to close the project or quit Audacity, no prompt to save the changes is triggered.

**Comment by chinakov** [comment]
Tested, here's an issue I've noticed. In specs, the clip name is shown in the dialog window title, like so: 
![image](https://github.com/user-attachments/assets/8b5dfbeb-577e-446e-82e4-2407db2e6db1)


Regarding previous issues:
1. Changes in pitch or speed are now real-time.
2. A progress bar is now shown for rendering clip pitch.
3. Clips' pitch can now be reset via shortcut clicking the pitch icon on the clip.
4. Clip's pitch change can be undone/redone.
5. Audacity UI is now active during opened pitch window.
6. Values are now validated.
7. Changing the clips speed now makes the project dirty.
